src/unix/apple/macosx/aes-gcm.c

```diff
@@ -31,17 +31,23 @@ struct MTY_AESGCM {
 
 MTY_AESGCM *MTY_AESGCMCreate(const void *key)
 {
+	size_t key_len = strlen(key);
+	if (key_len != 16 && key_len != 32) {
```

Comment:
a hexadecimal key would be 32 or 64 bytes. a 16 or 32 byte key is binary, in which case are we sure that NULL can't appear in the key itself? far less important, but worth mentioning: matoya/parsec is written as if malloc/free/strlen are 1-cycle machine instructions, when in fact they are usually the cause of performance death by 1000 papercuts. high performance code avoids strlen and passes a (void *begin, void *end) or a (void *begin, int count) to avoid calling strlen over and over. in this case strlen is relatively harmless, but a strlen habit is worth breaking.

Reply:
one thing that most dev teams would do, is to make a `struct key { uint8_t value[32]; uint32_t length; };` and pass it in, and maybe have a function `uint32_t key_length(const struct key*)` that returns `length,` or something like that. the parsec-ism of ripping the scalars out of 4 structs and passing them as 10 float/int parameters to a function, in some ways can reduce vexatious dependencies, but limits the abstractional power of the language, which is already pretty low in C